src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -4446,6 +4446,19 @@ public static void MaxSizes()
             Assert.Equal(0x7FFFFFC7, Array.MaxLength);
         }
 
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtimelab/issues/155", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))] // Array.Initialize
```

Comment:
It should not be hard to implement this for native AOT:

```
    EETypePtr pElementEEType = ElementEEType;
    if (!pElementEEType.IsValueType)
        return;

    delegate* <ref byte, void> constructorEntryPoint = (delegate* <ref byte, void>)
        RuntimeAugments.TypeLoaderCallbacks.TryGetDefaultConstructorForType(new RuntimeTypeHandle(pElementEEType))
    if (constructorEntryPoint == null)
        return;

    ref byte arrayRef = ref MemoryMarshal.GetArrayDataReference(this);
    nuint elementSize = ElementSize;

    for (int i = 0; i < Length; i++)
    {
        constructorEntryPoint(ref arrayRef);
        arrayRef = ref Unsafe.Add(ref arrayRef, elementSize);
    }
```

Reply:
Excellent, I tried looking for something like TryGetDefaultConstructorForType but couldn't quite find it. Thanks, I'll put that in.